app/src/main/java/com/gathering/android/auth/repo/ApiAuthRepository.kt

```diff
@@ -64,78 +57,40 @@ class ApiAuthRepository @Inject constructor(
         }
     }
 
-    override fun sendEmailVerification(
-        email: String, onResponseReady: (ResponseState<String>) -> Unit
-    ) {
-        remoteService.sendEmailVerification(email = email)
-            .enqueue(object : Callback<GeneralApiResponse> {
-                override fun onResponse(
-                    call: Call<GeneralApiResponse>, response: Response<GeneralApiResponse>
-                ) {
-                    if (!response.isSuccessful) {
-                        onResponseReady(ResponseState.Failure(Exception(RESPONSE_IS_NOT_SUCCESSFUL)))
-                        return
-                    }
-                    onResponseReady(ResponseState.Success(response.body()?.message ?: ""))
-                }
-
-                override fun onFailure(call: Call<GeneralApiResponse>, t: Throwable) {
-                    onResponseReady(ResponseState.Failure(t))
-                }
-            })
-    }
-
-    override suspend fun sendEmailVerification1(email: String) {
-        remoteService.sendEmailVerification1(email = email)
-    }
-
-    override fun emailVerify(
-        token: String, onResponseReady: (ResponseState<AuthorizedResponse>) -> Unit
-    ) {
-        remoteService.emailVerify(token).enqueue(object : Callback<AuthorizedResponse> {
-            override fun onResponse(
-                call: Call<AuthorizedResponse>, response: Response<AuthorizedResponse>
-            ) {
-                if (!response.isSuccessful) {
-                    onResponseReady(
-                        ResponseState.Failure(
-                            Exception(
-                                RESPONSE_IS_NOT_SUCCESSFUL
-                            )
-                        )
-                    )
-                    return
-                }
-                response.body()?.also { body ->
-                    userRepo.saveUser(body.user)
-                    tokenRepo.saveToken(body.jwt)
-                    onResponseReady(ResponseState.Success(body))
-                } ?: run {
-                    onResponseReady(
-                        ResponseState.Failure(
-                            Exception(
-                                RESPONSE_IS_NOT_SUCCESSFUL
-                            )
-                        )
-                    )
-                }
-            }
-
-            override fun onFailure(call: Call<AuthorizedResponse>, t: Throwable) {
-                onResponseReady(ResponseState.Failure(t))
+    override suspend fun sendEmailVerification(email: String) {
+        try {
+            remoteService.sendEmailVerification(
+                email = email,
+            )
+        } catch (e: HttpException) {
+            val throwable = when (e.code()) {
+                BAD_REQUEST_HTTP_CODE -> AuthException.FailedConnectingToServerException
```

Comment:
The BAD request is not "failed connecting to server", it means something in the request body or query parameter that you are sending to the server is not right, please correct this by throwing another exception

Reply:
done